File: interop/simple_test.cpp
#include <zeroipc/memory.h>
#include <zeroipc/array.h>
#include <zeroipc/queue.h>
#include <iostream>
#include <cstring>

// Simple structure compatible with Python
struct DataPoint {
    float value;
    uint32_t id;
};

int main() {
    std::cout << "=== Simple C++ Interop Test ===" << std::endl;

    // Open existing memory created by Python
    zeroipc::Memory mem("/simple_interop");

    // Access array created by Python
    zeroipc::Array<float> arr(mem, "float_array");
    std::cout << "Found float array with capacity: " << arr.capacity() << std::endl;

    // Read values
    for (size_t i = 0; i < 5 && i < arr.capacity(); i++) {
        std::cout << "  [" << i << "] = " << arr[i] << std::endl;
    }

    // Modify a value
    arr[0] = 99.99f;
    std::cout << "Modified arr[0] to 99.99" << std::endl;

    // Access queue
    zeroipc::Queue<int32_t> queue(mem, "int_queue");
    std::cout << "Found queue with size: " << queue.size() << std::endl;

    // Add some values
    queue.push(100);
    queue.push(200);
    std::cout << "Added values 100, 200 to queue" << std::endl;

    std::cout << "C++ test complete" << std::endl;
    return 0;
}

File: interop/simple_test.py
#!/usr/bin/env python3
"""Simple interop test - Python side"""

import sys
sys.path.insert(0, '/home/spinoza/github/beta/zeroipc/python')

import numpy as np
import time
from zeroipc import Memory, Array, Queue

def main():
    print("=== Simple Python Interop Test ===")

    # Create shared memory
    mem = Memory("/simple_interop", 1024 * 1024)  # 1MB
    print(f"Created shared memory: {mem}")

    # Create float array
    arr = Array(mem, "float_array", capacity=10, dtype=np.float32)
    for i in range(10):
        arr[i] = float(i * 1.1)
    print(f"Created float array with {len(arr)} elements")
    print(f"Initial values: {arr[:5]}")

    # Create int queue
    queue = Queue(mem, "int_queue", capacity=20, dtype=np.int32)
    for i in range(5):
        queue.push(i * 10)
    print(f"Created queue with {queue.size()} elements")

    print("\nWaiting for C++ to modify...")
    time.sleep(2)

    # Check modifications
    print(f"\nAfter C++ modifications:")
    print(f"arr[0] = {arr[0]} (should be 99.99)")
    print(f"Queue size: {queue.size()}")

    # Pop all queue elements
    print("Queue contents:")
    while not queue.empty():
        val = queue.pop()
        print(f"  Popped: {val}")

    print("\nPython test complete")

if __name__ == "__main__":
    main()